lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java

```diff
@@ -431,4 +436,209 @@ public void shouldAcceptCustomArrayClaimOfTypeLong() throws Exception {
+
+        Map<String, Object> sub = new HashMap<>();
+        sub.put("subKey", "subValue");
+        
+        data.add(sub);
+
+        String jwt = JWTCreator.init()
+                .withClaim("data", data)
+                .sign(Algorithm.HMAC256("secret"));
+
+        assertThat(jwt, is(notNullValue()));
+        String[] parts = jwt.split("\\.");
+        
+        String body = new String(Base64.decodeBase64(parts[1]), StandardCharsets.UTF_8);
+        ObjectMapper mapper = new ObjectMapper();
+        List<Object> list = (List<Object>) mapper.readValue(body, Map.class).get("data");
+        
+        assertThat(list.get(0), is("abc"));
+        assertThat(list.get(1), is(1));
+        assertThat(list.get(2), is(Long.MAX_VALUE));
+        assertThat(list.get(3), is(123.456d));
+        assertThat(list.get(4), is(123));
+        assertThat(list.get(5), is(true));
+        
+        // array types
+        assertThat(list.get(6), is(Arrays.asList(new Integer[]{3, 5})));
+        assertThat(list.get(7), is(Arrays.asList(new Long[]{Long.MAX_VALUE, Long.MIN_VALUE})));
+        assertThat(list.get(8), is(Arrays.asList(new String[]{"string"})));
+
+        // list
+        assertThat(list.get(9), is(Arrays.asList("a", "b", "c")));
+        assertThat(list.get(10), is(sub));
+
+    }
+    
+    @Test
+    public void shouldAcceptCustomClaimForNullListItem() throws Exception{
+        Map<String, Object> data = new HashMap<>();
+        data.put("test1", Arrays.asList("a", null, "c"));
+        
+        JWTCreator.init()
+                 .withClaim("pojo", data)
+                 .sign(Algorithm.HMAC256("secret"));
+    }
+    
+    @Test
+    public void shouldRefuseCustomClaimForNullMapValue() throws Exception{
+        Map<String, Object> data = new HashMap<>();
+        data.put("subKey", null);
+        
+        exception.expect(IllegalArgumentException.class);
+
+        JWTCreator.init()
+                 .withClaim("pojo", data)
+                 .sign(Algorithm.HMAC256("secret"));
+    }
+    
+    @Test
+    public void shouldRefuseCustomClaimForNullMapKey() throws Exception{
+        Map<String, Object> data = new HashMap<>();
+        data.put(null, "subValue");
+        
+        exception.expect(IllegalArgumentException.class);
+
+        JWTCreator.init()
+                 .withClaim("pojo", data)
+                 .sign(Algorithm.HMAC256("secret"));
+    }    
+    
+    @Test
+    public void shouldRefuseCustomMapClaimForNonStringKey() throws Exception{
+        Map data = new HashMap<>();
+        data.put(new Object(), "value");
+        
+        exception.expect(IllegalArgumentException.class);
+
+        JWTCreator.init()
+                 .withClaim("pojo", (Map<String, Object>)data)
+                 .sign(Algorithm.HMAC256("secret"));
+    }
+    
+    @Test
+    public void shouldRefuseCustomListClaimForUnknownListElement() throws Exception{
+    	List<Object> list = Arrays.asList(new UserPojo("Michael", 255));
```

Comment:
Correct. However, if these contents were written into a Map first, the List would have accepted that. Right? Do you think it's worth testing that as well?

Reply:
Adding another test